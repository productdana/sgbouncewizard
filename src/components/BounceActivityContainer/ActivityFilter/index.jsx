import React from "react";
import { Button } from "@sendgrid/ui-components/button";
import { TextInput } from "@sendgrid/ui-components/text-input";
import { Select } from "@sendgrid/ui-components/select";
import { Row } from "../../Row";
import { Column } from "../../Column";
import "./index.scss";

const ActivityFilter = ({
  searchToken,
  updateSearchToken,
  updateSearchCategory,
  filterOptions,
  addFilter,
  removeFilter,
  isValidFilter,
}) => (
  <div className="filter-wrap">
    <div className="filter-header">
      <p className="filter-title">
        {"Search activity by "}
        <a className="filter-option-toggle" href="#">
          all
        </a>
        {" of the following:"}
      </p>
      <div className="btn-list">
        <Button type="secondary" small>
          Clear Search
        </Button>
      </div>
    </div>
    <div className="filter-list">
      <Row>
        <Column width={3} offset={1}>
          <Select
            className="select-filter"
            onChange={updateSearchCategory}
            name="primary-filter"
            defaultValue={{ label: "Bounce Action", value: "Bounce Action" }}
            options={[
              { label: "Bounce Action", value: "Bounce Action" },
              { label: "Enhanced Code", value: "Enhanced Code" },
              { label: "Description", value: "description" },
            ]}
          />
        </Column>
        <Column width={9} offset={4}>
          <TextInput
            type="text"
            fullWidth
            label="Description"
            onChange={updateSearchToken}
            value={searchToken}
            isValid={isValidFilter}
            info={!isValidFilter && "Please enter a valid filter description."}
          />
        </Column>
        <i className="filter-row-remove sg-icon sg-icon-x" />
      </Row>

      <Row>
        <div className="row btn-list filter-actions-container">
          <Button type="primary" small onClick={addFilter}>
            Add a Filter
          </Button>
          {filterOptions.map(filterOption => (
            <Button
              type="secondary"
              small
              onClick={removeFilter}
              key={filterOption.searchCategory + filterOption.searchToken}
            >
              {filterOption.searchCategory} - {filterOption.searchToken}
            </Button>
          ))}
        </div>
      </Row>
    </div>
  </div>
);
export default ActivityFilter;
