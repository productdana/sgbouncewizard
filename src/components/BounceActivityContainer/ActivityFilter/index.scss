.filter-field-container {
  padding-bottom: 15px;
}

.filter-actions-container {
  padding-top: 15px;
}

.select-filter {
  padding-top: 15px;
}
