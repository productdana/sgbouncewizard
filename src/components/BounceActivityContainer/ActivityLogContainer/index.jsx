import React from "react";
import {
  HeaderCell,
  TableCell,
  Table,
  TableBody,
  TableHeader,
  TableRow,
} from "@sendgrid/ui-components/table/table";
import { shouldDisplay, displayUnixAsTime } from "../../../utils/utils";
import "./index.scss";

const ActivityLogContainer = ({ activityLog, handleActivityClicked }) => (
  <Table className="activity-table">
    <TableHeader>
      <TableRow>
        <HeaderCell className="row-date">Date</HeaderCell>
        <HeaderCell className="row-user">User</HeaderCell>
        <HeaderCell className="row-activity">Activity</HeaderCell>
        <HeaderCell className="row-message">Commit Message</HeaderCell>
        <HeaderCell className="actions-cell row-action">Actions</HeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody>
      {activityLog.map((log, index) => (
        <BounceRuleMin
          handleActivityClicked={handleActivityClicked}
          key={log.created_at}
          log={log}
          index={index}
        />
      ))}
    </TableBody>
  </Table>
);

const BounceRuleMin = ({ log, index, handleActivityClicked }) => {
  const {
    id,
    created_at: createdAt,
    user_id: userId,
    operation,
    comment,
  } = log;
  return (
    <TableRow data-id={id} data-time={createdAt}>
      <TableCell>{shouldDisplay(displayUnixAsTime(createdAt))}</TableCell>
      <TableCell>{shouldDisplay(userId)}</TableCell>
      <TableCell className={operation.toLowerCase()}>
        {shouldDisplay(operation)}
      </TableCell>
      <TableCell>{shouldDisplay(comment)}</TableCell>
      <TableCell className="changelog-view-icon-cell">
        <i
          onClick={handleActivityClicked}
          onKeyDown={handleActivityClicked}
          className="sg-icon sg-icon-view changelog-view-icon"
          id="isActivityModalOpen"
          rule-id={id}
          role="button"
          index={index}
          tabIndex={0}
        />
      </TableCell>
    </TableRow>
  );
};

export default ActivityLogContainer;
