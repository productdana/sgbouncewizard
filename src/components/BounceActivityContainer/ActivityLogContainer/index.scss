.activity-table {
  table-layout: fixed;

  .row-id {
    width: 5%;
  }

  .row-date {
    width: 15%;
  }

  .row-user {
    width: 10%;
  }

  .row-activity {
    width: 25%;
  }

  .row-message {
    width: 30%;
  }

  .row-action {
    width: 10%;
  }

  .new {
    color: green;
  }

  .create {
    color: green;
  }

  .delete {
    color: red;
  }

  .update {
    color: #2196f3;
  }
}
