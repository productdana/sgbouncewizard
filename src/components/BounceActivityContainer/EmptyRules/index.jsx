import React from "react";
import EmptyState from "@sendgrid/ui-components/empty-state";

const EmptyRules = () => (
  <EmptyState icon="warning-triangle" header="No rules available.">
    <p>No rules are currently available to display.</p>
  </EmptyState>
);

export default EmptyRules;
