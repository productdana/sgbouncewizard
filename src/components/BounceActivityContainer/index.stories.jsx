import React from "react";
import { storiesOf } from "@storybook/react";
import StoryRouter from "storybook-react-router";
import BounceRulesActivity from "./index";

const log1 = {
  bounce_action: "cypressDeleteTest",
  comment: "",
  created_at: 1550348178,
  description: "testDescription",
  enhanced_code: "492",
  id: 670,
  operation: "Delete",
  priority: 1,
  regex: "testRegex",
  response_code: 528,
  user_id: 2,
};

const log2 = {
  bounce_action: "cypressTestTest",
  comment: "",
  created_at: 1550453178,
  description: "testDescription",
  enhanced_code: "492",
  id: 671,
  operation: "Create",
  priority: 1,
  regex: "testRegex",
  response_code: 528,
  user_id: 2,
};

const log3 = {
  bounce_action: "cypressTestTest",
  comment: "",
  created_at: 1550453278,
  description: "testDescription",
  enhanced_code: "492",
  id: 671,
  operation: "Update",
  priority: 1,
  regex: "testRegex",
  response_code: 528,
  user_id: 2,
};

storiesOf("Bounce Activity Page", module)
  .addDecorator(StoryRouter())
  .add("Default", () => (
    <BounceRulesActivity
      activityLog={[log1, log2, log3]}
      filteredActivityLog={[log1, log2, log3]}
      filterOptions={[]}
      isActivityLogTab
    />
  ))
  .add("Empty", () => (
    <BounceRulesActivity
      activityLog={[]}
      filteredActivityLog={[]}
      filterOptions={[]}
      isActivityLogTab
    />
  ))
  .add("View Details", () => (
    <BounceRulesActivity
      activityLog={[log1, log2, log3]}
      filteredActivityLog={[log1, log2, log3]}
      filterOptions={[]}
      isActivityLogTab
      isActivityModalOpen
      selectedActivity={log1}
    />
  ));
