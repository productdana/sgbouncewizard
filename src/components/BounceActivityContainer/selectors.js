import {
  selectSelectorGenerator,
  writeSelectorGenerator,
} from "../selectorHelpers";

// These test hooks will eventually turn into the following:
// - Selectors (CSS Selector strings) i.e. "[data-test=emailInput]"
// - Write Selectors (Selector objects spread onto React components) i.e. { data-test: "emailInput" }
// What if we had to update a data-test attribute value or other CSS selectors?
// - We would have to update all the test file strings in Jest/Enzyme and Cypress tests and React components.
// - This pattern helps us maintain this in the long run and update as few places as possible for CSS selector changes
const testHooks = {
  page: "activity-log-page",
  breadcrumb: "activity-log-breadcrumb",
  csvButton: "export-csv-button",
  activityFilter: "rule-filter",
  activityTable: "activity-table",
  emptyRulesWarning: "empty-rules-warning",
  pagination: "pagination",
  detailsModal: "actvity-details-modal",
};

// Selectors like "[data-test=emailInput]" to be imported into the following files:
// - Jest/Enzyme tests to be placed into .find("[data-test=emailInput]") and other similar functions
// - Cypress tests to be placed into cy.get("[data-test=emailInput]"), cy.contains(), etc.
const Selectors = selectSelectorGenerator(testHooks);

// Write Selectors like { emailInput: { data-test: "emailInput" }, ... } to be imported into
// React component files to spread the data-test attributes onto elements we would like to keep track of for tests
// i.e. <Component {...WriteSelectors.emailInput} /> => <Component data-test="emailInput" />
const WriteSelectors = writeSelectorGenerator(testHooks);

export { Selectors, WriteSelectors };
