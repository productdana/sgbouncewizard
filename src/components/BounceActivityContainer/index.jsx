import React from "react";
import PropTypes from "prop-types";
import { Redirect } from "react-router";
import { Link } from "react-router-dom";
import { CSVLink } from "react-csv";
import Breadcrumb from "@sendgrid/ui-components/breadcrumb";
import { StatefulTabs as Tabs, Tab } from "@sendgrid/ui-components/tabs";
import Loader from "@sendgrid/ui-components/loader";
import Header from "../Header";
import { Row } from "../Row";
import { Column } from "../Column";
import Pagination from "../Pagination";
import ActivityFilter from "./ActivityFilter";
import EmptyRules from "./EmptyRules";
import ActivityLogContainer from "./ActivityLogContainer";
import ActivityDetailsModal from "./Modals/ActivityDetailsModal";
import "./index.scss";
import { WriteSelectors } from "./selectors";

const BounceActivityContainer = ({
  updateSearchToken,
  updateSearchCategory,
  removeFilter,
  filteredActivityLog,
  searchToken,
  rulesToShow,
  pagesToDisplay,
  filterOptions,
  addFilter,
  isValidFilter,
  handleModalClose,
  isFetching,
  logout,
  handleActivityTabClicked,
  handleBounceTabClicked,
  isBounceRulesTab,
  isActivityLogTab,
  activityLog,
  updatePageIndex,
  handlePrevClicked,
  handleNextClicked,
  currentPageIndex,
  handleActivityClicked,
  selectedActivity,
  isActivityModalOpen,
}) => {
  const isActivityEmpty = activityLog.length === 0;
  const shouldShowActivityLogPagination =
    isActivityLogTab && !isActivityEmpty && !isFetching;
  const shouldShowActivityLogContainer = isActivityLogTab && !isActivityEmpty;
  const shouldShowEmpty = isActivityEmpty && !isFetching;
  return (
    <React.Fragment>
      {isBounceRulesTab && <Redirect push to="/bounce_rules" />}
      <div {...WriteSelectors.page}>
        <Header logout={logout} />
        <Row>
          <Column width={6} offset={2}>
            <Breadcrumb>
              <Link to="/activity_log"> Activity Log</Link>
            </Breadcrumb>
          </Column>
        </Row>
        <Row>
          <Column className="csv-button-col" width={4} offset={8}>
            <CSVLink
              {...WriteSelectors.csvButton}
              filename="activity_log.csv"
              className="sg-button btn btn-secondary sg-right"
              data={activityLog}
            >
              Export CSV
            </CSVLink>
          </Column>
        </Row>
        <Row>
          <Column width={10} offset={2}>
            <Tabs className="page-tab" onChange={() => {}}>
              <Tab onClick={handleBounceTabClicked}>Bounce Rules</Tab>
              <Tab onClick={handleActivityTabClicked} active>
                Activity Log
              </Tab>
            </Tabs>
          </Column>
        </Row>
        <Row>
          <Column width={10} offset={2}>
            <div {...WriteSelectors.activityFilter}>
              <ActivityFilter
                searchToken={searchToken}
                updateSearchToken={updateSearchToken}
                updateSearchCategory={updateSearchCategory}
                filterOptions={filterOptions}
                addFilter={addFilter}
                removeFilter={removeFilter}
                isValidFilter={isValidFilter}
              />
            </div>
          </Column>
        </Row>
        <Row>
          <Column width={10} offset={2}>
            {isFetching && (
              <div className="bounce-rule-loader">
                <Loader centered />
                <Row>
                  <p className="loading">Loading Rules</p>
                </Row>
              </div>
            )}
            {shouldShowEmpty && (
              <div {...WriteSelectors.emptyRulesWarning}>
                <EmptyRules />
              </div>
            )}
            {shouldShowActivityLogContainer && (
              <div {...WriteSelectors.activityTable}>
                <ActivityLogContainer
                  activityLog={filteredActivityLog}
                  handleActivityClicked={handleActivityClicked}
                />
              </div>
            )}
          </Column>
        </Row>
        <Row>
          {shouldShowActivityLogPagination && (
            <Pagination
              handlePrevClicked={handlePrevClicked}
              handleNextClicked={handleNextClicked}
              currentPageIndex={currentPageIndex}
              rulesToShow={rulesToShow}
              numRules={activityLog.length}
              updatePageIndex={updatePageIndex}
              pagesToDisplay={pagesToDisplay}
            />
          )}
        </Row>
        {isActivityModalOpen && (
          <ActivityDetailsModal
            handleModalClose={handleModalClose}
            selectedChange={selectedActivity}
          />
        )}
      </div>
    </React.Fragment>
  );
};

BounceActivityContainer.propTypes = {
  updateSearchToken: PropTypes.func,
  updateSearchCategory: PropTypes.func,
  removeFilter: PropTypes.func,
  handlePrevClicked: PropTypes.func,
  handleNextClicked: PropTypes.func,
  updatePageIndex: PropTypes.func,
  searchToken: PropTypes.string,
  currentPageIndex: PropTypes.number,
  rulesToShow: PropTypes.number,
  pagesToDisplay: PropTypes.number,
  filterOptions: PropTypes.arrayOf(
    PropTypes.shape({
      searchCategory: PropTypes.string,
      searchToken: PropTypes.string,
    })
  ),
  addFilter: PropTypes.func,
  isValidFilter: PropTypes.bool,
  isCreateRuleOpen: PropTypes.bool,
  handleModalClose: PropTypes.func,
  isFetching: PropTypes.bool,
  numRules: PropTypes.number,
};

BounceActivityContainer.defaultProps = {
  updateSearchToken: () => {},
  updateSearchCategory: () => {},
  removeFilter: () => {},
  handlePrevClicked: () => {},
  handleNextClicked: () => {},
  updatePageIndex: () => {},
  searchToken: "",
  currentPageIndex: 1,
  rulesToShow: 10,
  pagesToDisplay: 5,
  numRules: 0,
  filterOptions: () => {},
  addFilter: () => {},
  isCreateRuleOpen: false,
  handleModalClose: () => {},
  isFetching: false,
  isValidFilter: true,
};

export default BounceActivityContainer;
