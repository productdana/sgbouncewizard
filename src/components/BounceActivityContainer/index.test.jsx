import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow } from "enzyme";
import BounceActivityContainer from ".";
import { Selectors } from "./selectors";
import mockChangelog from "../../mocks/index";

const {
  csvButton,
  activityFilter,
  activityTable,
  emptyRulesWarning,
  pagination,
} = Selectors;

describe("Bounce Activity Page", () => {
  let props;
  let mountedBounceActivityPage;
  const BounceActivityPage = () => {
    if (!mountedBounceActivityPage) {
      mountedBounceActivityPage = shallow(
        <BounceActivityContainer {...props} />
      );
    }
    return mountedBounceActivityPage;
  };

  beforeEach(() => {
    props = {
      activityLog: mockChangelog,
      filteredActivityLog: mockChangelog,
      filterOptions: [],
      isFetching: false,
    };
    mountedBounceActivityPage = undefined;
  });

  it("should render correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <BounceActivityContainer {...props} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render a export to csv button", () => {
    expect(BounceActivityPage().find(csvButton)).toHaveLength(1);
  });

  it("should render a filter component", () => {
    expect(BounceActivityPage().find(activityFilter)).toHaveLength(1);
  });

  it("should render a activity table component", () => {
    expect(BounceActivityPage().find(activityTable)).toHaveLength(1);
  });

  it("should render paginiation", () => {
    expect(BounceActivityPage().find(pagination)).toHaveLength(1);
  });

  it("should render warning when no rules available", () => {
    BounceActivityPage().setProps({ activityLog: [] });
    expect(BounceActivityPage().find(emptyRulesWarning)).toHaveLength(1);
  });
});
