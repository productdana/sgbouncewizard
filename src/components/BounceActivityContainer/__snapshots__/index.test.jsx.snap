// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bounce Activity Page should render correctly 1`] = `
<div
  data-test="activity-log-page"
>
  <div
    className="header-container"
  >
    <div
      className="mako-grid__row"
    >
      <div
        className="mako-grid__col-3 col-start-2"
      >
        <img
          alt="sendgrid-logo"
          className="sg-logo"
          data-test="sg-logo"
          src="https://uiux.s3.amazonaws.com/toggleable-logos/header-logo.svg"
          style={
            Object {
              "maxWidth": "150px",
            }
          }
        />
      </div>
      <div
        className="mako-grid__col-2 col-start-10"
      >
        <div
          className="header-user-container"
        >
          <span
            className="header-greeting"
            data-test="user-greeting"
          >
            Hello, 
            User
            !
          </span>
          <a
            className="header-logout"
            data-test="logout-button"
            role="button"
            tabIndex={0}
          >
            Logout
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-6 col-start-2"
    >
      <ol
        className="breadcrumb"
      >
        <li>
          <a
            href="/activity_log"
            onClick={[Function]}
          >
             Activity Log
          </a>
        </li>
        <li />
      </ol>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <ul
        className="tab-wrapper page-tab"
      >
        <li
          className="tab"
          onClick={[Function]}
        >
          Bounce Rules
           
        </li>
        <li
          className="tab undefined is-active"
          onClick={[Function]}
        >
          Activity Log
           
        </li>
      </ul>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-4 col-start-8 csv-button-col"
    >
      <a
        className="sg-button btn btn-secondary sg-right"
        data-test="export-csv-button"
        download="activity_log.csv"
        href="data:text/csv;charset=utf-8,ï»¿\\"id\\",\\"response_code\\",\\"enhanced_code\\",\\"regex\\",\\"priority\\",\\"description\\",\\"bounce_action\\",\\"user_id\\",\\"created_at\\",\\"comment\\",\\"operation\\"
\\"173\\",\\"421\\",\\"sdfsfdfadf\\",\\"fewgfgsdfg\\",\\"1\\",\\"RFC5321Servicenotavailable\\",\\"retry\\",\\"1\\",\\"1\\",\\"Inital Setup\\",\\"Create\\"
\\"174\\",\\"450\\",\\"sadfasdf\\",\\"asdfsdfd\\",\\"0\\",\\"RFC5321Mailboxunavailable\\",\\"retry\\",\\"1\\",\\"2\\",\\"Inital Setup\\",\\"Create\\"
\\"175\\",\\"451\\",\\"sdfsdf\\",\\"sdfasf\\",\\"1\\",\\"RFC5321Localerrorinprocessing\\",\\"retry\\",\\"1\\",\\"3\\",\\"Inital Setup\\",\\"Create\\"
\\"176\\",\\"452\\",\\"sdfgerg\\",\\"dsfgsdfg\\",\\"1\\",\\"RFC5321Insufficientsystemstorage\\",\\"retry\\",\\"1\\",\\"4\\",\\"Inital Setup\\",\\"Create\\"
\\"177\\",\\"454\\",\\"sdfgsdfg\\",\\"dfgdfg\\",\\"1\\",\\"RFC3207TLSnotavailable\\",\\"retry\\",\\"1\\",\\"5\\",\\"Inital Setup\\",\\"Create\\"
\\"178\\",\\"455\\",\\"sdfgreg\\",\\"sdfgsdfg\\",\\"0\\",\\"RFC5321Serverunabletoaccomodateparameters\\",\\"retry\\",\\"1\\",\\"6\\",\\"Inital Setup\\",\\"Create\\"
\\"179\\",\\"500\\",\\"dfgsdfgre\\",\\"dsfgsd\\",\\"0\\",\\"RFC5321Syntaxerrorcommandunrecognized\\",\\"no_action\\",\\"1\\",\\"7\\",\\"Inital Setup\\",\\"Create\\"
\\"180\\",\\"501\\",\\"sdfgdsfg\\",\\"dfgsfg\\",\\"1\\",\\"RFC5321Syntaxerrorinparametersorarguments\\",\\"no_action\\",\\"1\\",\\"8\\",\\"Inital Setup\\",\\"Create\\"
\\"181\\",\\"502\\",\\"fdgert\\",\\"sdfg\\",\\"1\\",\\"RFC5321Commandnotimplemented\\",\\"no_action\\",\\"1\\",\\"9\\",\\"Inital Setup\\",\\"Create\\""
        onClick={[Function]}
        target="_self"
      >
        Export CSV
      </a>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <div
        data-test="rule-filter"
      >
        <div
          className="filter-wrap"
        >
          <div
            className="filter-header"
          >
            <p
              className="filter-title"
            >
              Search activity by 
              <a
                className="filter-option-toggle"
                href="#"
              >
                all
              </a>
               of the following:
            </p>
            <div
              className="btn-list"
            >
              <button
                className="btn btn-secondary btn-small"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Clear Search
              </button>
            </div>
          </div>
          <div
            className="filter-list"
          >
            <div
              className="mako-grid__row"
            >
              <div
                className="mako-grid__col-3 col-start-1"
              >
                <div
                  className="input-select-wrap"
                >
                  <div
                    className="css-5jw5i7 select-filter"
                    onKeyDown={[Function]}
                  >
                    <div
                      className="css-12nb859"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className="css-1m8oe7c"
                      >
                        <div
                          className="css-1v8euj5"
                        >
                          Bounce Action
                        </div>
                        <div
                          className="css-1g6gooi"
                        >
                          <div
                            className=""
                            style={
                              Object {
                                "display": "inline-block",
                              }
                            }
                          >
                            <input
                              aria-autocomplete="list"
                              autoCapitalize="none"
                              autoComplete="off"
                              autoCorrect="off"
                              disabled={false}
                              id="react-select-2-input"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              spellCheck="false"
                              style={
                                Object {
                                  "background": 0,
                                  "border": 0,
                                  "boxSizing": "content-box",
                                  "color": "inherit",
                                  "fontSize": "inherit",
                                  "opacity": 1,
                                  "outline": 0,
                                  "padding": 0,
                                  "width": "1px",
                                }
                              }
                              tabIndex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style={
                                Object {
                                  "height": 0,
                                  "left": 0,
                                  "overflow": "scroll",
                                  "position": "absolute",
                                  "top": 0,
                                  "visibility": "hidden",
                                  "whiteSpace": "pre",
                                }
                              }
                            >
                              
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        className="css-1wy0on6"
                      >
                        <span
                          className="css-0"
                        />
                        <div
                          aria-hidden="true"
                          className="css-86sq3t"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <svg
                            aria-hidden="true"
                            className="css-19bqh2r"
                            focusable="false"
                            height={20}
                            viewBox="0 0 20 20"
                            width={20}
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="primary-filter"
                      type="hidden"
                      value="Bounce Action"
                    />
                  </div>
                </div>
              </div>
              <div
                className="mako-grid__col-9 col-start-4"
              >
                <div
                  className="input-text-wrap"
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <label
                    className="input-text-label"
                  >
                    Description
                  </label>
                  <input
                    aria-describedby={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    step={null}
                    type="text"
                    value=""
                  />
                  
                </div>
              </div>
              <i
                className="filter-row-remove sg-icon sg-icon-x"
              />
            </div>
            <div
              className="mako-grid__row"
            >
              <div
                className="row btn-list filter-actions-container"
              >
                <button
                  className="btn btn-primary btn-small"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Add a Filter
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <div
        data-test="activity-table"
      >
        <table
          className="table-wrap activity-table"
        >
          <thead>
            <tr>
              <th
                className="row-date"
                onClick={[Function]}
              >
                Date
              </th>
              <th
                className="row-user"
                onClick={[Function]}
              >
                User
              </th>
              <th
                className="row-activity"
                onClick={[Function]}
              >
                Activity
              </th>
              <th
                className="row-message"
                onClick={[Function]}
              >
                Commit Message
              </th>
              <th
                className="actions-cell row-action"
                onClick={[Function]}
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              data-id={173}
              data-time={1}
            >
              <td>
                12/31/1969 4:00:01 PM
              </td>
              <td>
                1
              </td>
              <td
                className="create"
              >
                Create
              </td>
              <td>
                Inital Setup
              </td>
              <td
                className="changelog-view-icon-cell"
              >
                <i
                  className="sg-icon sg-icon-view changelog-view-icon"
                  id="isActivityModalOpen"
                  index={0}
                  role="button"
                  rule-id={173}
                  tabIndex={0}
                />
              </td>
            </tr>
            <tr
              data-id={174}
              data-time={2}
            >
              <td>
                12/31/1969 4:00:02 PM
              </td>
              <td>
                1
              </td>
              <td
                className="create"
              >
                Create
              </td>
              <td>
                Inital Setup
              </td>
              <td
                className="changelog-view-icon-cell"
              >
                <i
                  className="sg-icon sg-icon-view changelog-view-icon"
                  id="isActivityModalOpen"
                  index={1}
                  role="button"
                  rule-id={174}
                  tabIndex={0}
                />
              </td>
            </tr>
            <tr
              data-id={175}
              data-time={3}
            >
              <td>
                12/31/1969 4:00:03 PM
              </td>
              <td>
                1
              </td>
              <td
                className="create"
              >
                Create
              </td>
              <td>
                Inital Setup
              </td>
              <td
                className="changelog-view-icon-cell"
              >
                <i
                  className="sg-icon sg-icon-view changelog-view-icon"
                  id="isActivityModalOpen"
                  index={2}
                  role="button"
                  rule-id={175}
                  tabIndex={0}
                />
              </td>
            </tr>
            <tr
              data-id={176}
              data-time={4}
            >
              <td>
                12/31/1969 4:00:04 PM
              </td>
              <td>
                1
              </td>
              <td
                className="create"
              >
                Create
              </td>
              <td>
                Inital Setup
              </td>
              <td
                className="changelog-view-icon-cell"
              >
                <i
                  className="sg-icon sg-icon-view changelog-view-icon"
                  id="isActivityModalOpen"
                  index={3}
                  role="button"
                  rule-id={176}
                  tabIndex={0}
                />
              </td>
            </tr>
            <tr
              data-id={177}
              data-time={5}
            >
              <td>
                12/31/1969 4:00:05 PM
              </td>
              <td>
                1
              </td>
              <td
                className="create"
              >
                Create
              </td>
              <td>
                Inital Setup
              </td>
              <td
                className="changelog-view-icon-cell"
              >
                <i
                  className="sg-icon sg-icon-view changelog-view-icon"
                  id="isActivityModalOpen"
                  index={4}
                  role="button"
                  rule-id={177}
                  tabIndex={0}
                />
              </td>
            </tr>
            <tr
              data-id={178}
              data-time={6}
            >
              <td>
                12/31/1969 4:00:06 PM
              </td>
              <td>
                1
              </td>
              <td
                className="create"
              >
                Create
              </td>
              <td>
                Inital Setup
              </td>
              <td
                className="changelog-view-icon-cell"
              >
                <i
                  className="sg-icon sg-icon-view changelog-view-icon"
                  id="isActivityModalOpen"
                  index={5}
                  role="button"
                  rule-id={178}
                  tabIndex={0}
                />
              </td>
            </tr>
            <tr
              data-id={179}
              data-time={7}
            >
              <td>
                12/31/1969 4:00:07 PM
              </td>
              <td>
                1
              </td>
              <td
                className="create"
              >
                Create
              </td>
              <td>
                Inital Setup
              </td>
              <td
                className="changelog-view-icon-cell"
              >
                <i
                  className="sg-icon sg-icon-view changelog-view-icon"
                  id="isActivityModalOpen"
                  index={6}
                  role="button"
                  rule-id={179}
                  tabIndex={0}
                />
              </td>
            </tr>
            <tr
              data-id={180}
              data-time={8}
            >
              <td>
                12/31/1969 4:00:08 PM
              </td>
              <td>
                1
              </td>
              <td
                className="create"
              >
                Create
              </td>
              <td>
                Inital Setup
              </td>
              <td
                className="changelog-view-icon-cell"
              >
                <i
                  className="sg-icon sg-icon-view changelog-view-icon"
                  id="isActivityModalOpen"
                  index={7}
                  role="button"
                  rule-id={180}
                  tabIndex={0}
                />
              </td>
            </tr>
            <tr
              data-id={181}
              data-time={9}
            >
              <td>
                12/31/1969 4:00:09 PM
              </td>
              <td>
                1
              </td>
              <td
                className="create"
              >
                Create
              </td>
              <td>
                Inital Setup
              </td>
              <td
                className="changelog-view-icon-cell"
              >
                <i
                  className="sg-icon sg-icon-view changelog-view-icon"
                  id="isActivityModalOpen"
                  index={8}
                  role="button"
                  rule-id={181}
                  tabIndex={0}
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <div
        className="pagination pagination-container"
        data-test="pagination"
      >
        <a
          className="btn btn-secondary btn-small pagination-prev is-invisible"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          Prev
        </a>
        <div
          className="pagination-links"
        >
          <a
            className="pagination-link is-active"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
            value={1}
          >
            1
          </a>
        </div>
        <a
          className="btn btn-secondary btn-small pagination-next is-invisible"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          Next
        </a>
      </div>
    </div>
  </div>
</div>
`;
